server/routes/api/browser.py

```diff
@@ -60,3 +61,53 @@ def get_property_labels(dcid):
     """Returns all property labels given a node dcid."""
     labels = dc.get_property_labels([dcid]).get(dcid, {})
     return Response(json.dumps(labels), 200, mimetype='application/json')
+
+
+@cache.cached(timeout=3600 * 24, query_string=True)  # Cache for one day.
+@bp.route('/observationId')
+def get_observation_id():
+    """Returns the dcid of the observation node for a combination of predicates: observedNodeLocation,
+    statisticalVariable, observationDate, measurementMethod (optional), observationPeriod (optional)"""
+    place_id = request.args.get("place")
+    if not place_id:
+        return Response(json.dumps("error: must provide a placeId field"),
+                        400,
+                        mimetype='application/json')
+    stat_var_id = request.args.get("stat_var")
+    if not stat_var_id:
+        return Response(json.dumps("error: must provide a statVarId field"),
+                        400,
+                        mimetype='application/json')
+    date = request.args.get("date")
+    if not date:
+        return Response(json.dumps("error: must provide a date field"),
+                        400,
+                        mimetype='application/json')
+    measurement_method = request.args.get("measurement_method", "")
+    observation_period = request.args.get("obs_period", "")
+    measurement_method_line = ""
+    if measurement_method:
+        measurement_method_line = "?observation measurementMethod " + measurement_method + " ."
+    observation_period_line = ""
+    if observation_period:
+        observation_period_line = "?observation observationPeriod " + observation_period + " ."
+    sparql_query = '''
```

Comment:
how long does this query take?

Reply:
this is the query that I mentioned could take up to 5 seconds